docs: added table of contents and introduction document for Studio tool (#553)
asyncapi/studio: docs/navigation.md

# Navigation panel

> This guide will walk you through the functions of each button in navigation panel. 

Reviewer: ```suggestion
This document will walk you through the functions of each button in the navigation panel. 
```

Reviewer: I wouldn't call it a guide because it won't be going under the `Guides` content bucket, and would prefer we avoid any confusion 😄

# Navigation Button

The first button in the navigation panel is the Navigation button. It is one of the default active buttons.

> Navigation button is divided into four parts. It provides the following information:

Reviewer: ```suggestion
The navigation button is divided into four parts. It provides the following information:
```

- Servers
- Operations
- Messages
- Schemas

## Servers

The Servers section displays the name and information of the [server]( https://www.asyncapi.com/docs/concepts/server) you are using to build your AsyncAPI.

## Operations

The Operations section lists all the operations that are a part of the AsyncAPI. It adds the perfix `PUB`  for publisher and `SUB` for subscriber to differentiate between both the operations.

## Messages

> This Messages section displays the messages stored in `components.messages` section. This does not mean that every message in the component is used.  It can be defined but it may not used by references in the `operation` field.

Reviewer: ```suggestion
The Messages section displays the messages stored in the `components.messages` section. This does not mean that every message in the component is used.  It can be defined but it may not used by references in the `operation` field.
```

## Schemas

The Schemas displays the schemas stored in the `components.schemas` section. This does not mean that every schema is used in the document. You can have it defined but it may not be used by references in the `message.payload` field.

# Editor

The Editor section is where you edit the AsyncAPI file. As you edit, you will see the changes reflect dynamically in the HTML preview and Block Visualizer section.

Click on the individual elements of the Navigation button to jump to those sections in HTML preview and Block Visualizer section.

## Diagnostics Window

> The Editor also has resizeable Diagnostics window. It shows a quick count of number of 

Reviewer: ```suggestion
The Editor also has a resizeable Diagnostics window. It shows a quick count number of... 
```

- errors (in red),
- warnings (in yellow),

> - informations (in blue),

Reviewer: ```suggestion
- information (in blue),
```

- hints (in green)

> by its title. The description of the diagnostics with line number is displayed in a table.

Reviewer: ```suggestion
...by its title. The description of the diagnostics with line number is displayed in a table.
```

To the right side of the title bar there are three indicators of the Diagnostics window: 

- Valid/Invalid - It is green if the syntax is green and red when invalid.
- Autosave - It shows whether Autosave is on or off.
- Type - Shows the language in which the API is written: `JSON` or `YAML`.

> Below the title bar there is a control panel. You can the buttons to:

Reviewer: ```suggestion
Below the title bar there is a control panel. You can use the buttons to:
```

- Hide and show erros
- Hide and show warnings
- Hide and show information messages
- Hide and show hints

> There is also search bar to search that aids in finding issues by their names. You will see Settings icon to the right. It will take you to the governance of Diagnostic's settings window.

Reviewer: ```suggestion
There is also a search bar that aids in finding issues by their names. You will see the Settings icon to the right. It will take you to the governance of Diagnostic's settings window.
```

# HTML Preview

This section displays the preview of everything you have written in the code editor. It uses different fonts, colors, and styles for readability. Each section has a different card. 

> It starts with displaying the title, version, description of the API. It renders the formatting too. You can expand each element too see the `key: value` pair of each element, type, payload, and message.

Reviewer: ```suggestion
It starts with displaying the title, version, and description of the API. It renders the formatting too. You can expand each element too via the `key: value` pair of each element, type, payload, and message.
```

# Blocks visualizer

Blocks visualizer gives you a quick overview of your API. Unlike HTML preview, Blocks visualizer gives a diagrammatic visual of your API. 

> Title card is at the top-left corner of the Block visualizer window. Followed by the Operations card.  You can visualize channels, published messages, payload, type,  patterns, and relationship between the components.

Reviewer: ```suggestion
The title card is at the top-left corner of the Block visualizer window. Followed by the Operations card.  You can visualize channels, published messages, payload, type,  patterns, and relationships between the components.
```

The Blocks visualizer has two set of controls:

- In the top right corner of the window you will find the play and reset button. Play to view live relationship between the components and its flow. Reset to get back to the default view.

> - To bottom left of the window there are four controls: zoom in, zoom out, full-screen, toogle interactivity. You zoom the cards in and out or view them in full-screen mode. To prevent the cards from shifting use the toggle interactivity control.

Reviewer: ```suggestion
- To the bottom left of the window there are four controls: zoom in, zoom out, full-screen, and toogle interactivity. You zoom the cards in and out or view them in full-screen mode. To prevent the cards from shifting use the toggle interactivity control.
```

## Templates

There are 5 available to quick start Studio. 

- Simple Example - A basic example of a service that is in charge of processing user signups. Great place to start learning AsyncAPI.

- Apache Kafka - A framework implementation of a software bus using stream-processing. Open Source developed by the Apache Software Foundation.

- WebSocket - A framework implementation of a software bus using stream-processing. Open Source developed by the Apache Software Foundation.

- MQTT - A protocol for fetching resources. It is the foundation of any data exchange on the Web and it is a client-server protocol.

- HTTP - A protocol for fetching resources. It is the foundation of any data exchange on the Web and it is a client-server protocol.

## Real World Examples

Studio provides two templates to help you get started.

- Slack Real Time Messaging API - Slack Real time messaging API. Using HTTP protocol.

- Gitter Streaming API - Gitter Streaming API from https://stream.gitter.im. Using HTTP protocol.

## Tutorials

Studio also provides you with a template of an invalid AsyncAPI document. The purpose of this tutorial is to educate, and learn document validation.

> [You can request to add a template](https://github.com/asyncapi/studio/issues/new?assignees=&labels=enhancement&template=enhancement.md&title=Template%20Request:%20{%20template%20name%20and%20type%20}) to the list if you don't find what you are looking for.

Reviewer: ```suggestion
[You can request to add a Studio template](https://github.com/asyncapi/studio/issues/new?assignees=&labels=enhancement&template=enhancement.md&title=Template%20Request:%20{%20template%20name%20and%20type%20}) to the list if you don't find what you are looking for.
```

# Navigation panel

This guide will walk you through the functions of each button in navigation panel. 

# Navigation Button

The first button in the navigation panel is the Navigation button. It is one of the default active buttons.

Navigation button is divided into four parts. It provides the following information:

- Servers
- Operations
- Messages
- Schemas

## Servers

The Servers section displays the name and information of the server you are using to build your AsyncAPI.

## Operations

The Operations section lists all the operations that are a part of the AsyncAPI. It adds the perfix `PUB`  for publisher and `SUB` for subscriber to differentiate between both the operations.

## Messages

This Messages section displays the title of each message used to communicate to the API.

## Schemas

The Schemas displays the title of the schemas defined.

# Editor

The Editor section is where you edit the AsyncAPI code. As you edit, you will see the changes reflect dynamically in the HTML preview and Block Visualizer section.

Click on the individual elements of the Navigation button to jump to those sections in HTML preview and Block Visualizer section.

## Diagnostics Window

> The Editor also has resizeable Diagnostics window. It shows a quick count of number of errors (in red) and warnings (in yellow) by its title. The description of the warnings and errors with line number is displayed in a table.

Reviewer 2: ```suggestion
The Editor also has resizeable Diagnostics window. It shows a quick count of number of 
- errors (in red),
- warnings (in yellow),
- informations (in blue),
- hints (in green)
by its title. The description of the diagnostics with line number is displayed in a table.
```

To the right side of the title bar there are three indicators of the Diagnostics window: 

- Valid/Invalid - It is green if the syntax is green and red when invalid.
- Autosave - It shows whether Autosave is on or off.
- Type - Shows the language in which the API is written: `JSON` or `YAML`.

> Bwlow the title bar there is a control panel. You can the buttons to:

Reviewer 2: ```suggestion
Below the title bar there is a control panel. You can the buttons to:
```

- Hide and show erros
- Hide and show warnings
- Hide and show information messages
- Hide and show hints

> There is also search bar to search that aids in finding issues by their names. You will see Settings icon to the right. It will take you to the governance of Diagnostic window.

Reviewer 2: ```suggestion
There is also search bar to search that aids in finding issues by their names. You will see Settings icon to the right. It will take you to the governance of Diagnostic's settings window.
```

# Blocks visualizer

Blocks visualizer is the USP of this product. It gives you a quick overview of your API. Unlike HTML preview, Blocks visualizer gives a diagrammatic visual of your API. 

Title card is at the top-left corner of the Block visualizer window. Followed by the Operations card.  You can visualize channels, published messages, payload, type,  patterns, and relationship between the components.

The Blocks visualizer has two set of controls:

- In the top right corner of the window you will find the play and reset button. Play to view live relationship between the components and its flow. Reset to get back to the default view.

- To bottom left of the window there are four controls: zoom in, zoom out, full-screen, toogle interactivity. You zoom the cards in and out or view them in full-screen mode. To prevent the cards from shifting use the toggle interactivity control.

# New File

To create a new file you can pick one of the available templates.

## Templates

There are 5 available to quick start Studio. 

> - _Simple Example - A basic example of a service that is in charge of processing user signups. Great place to start learning AsyncAPI.

Reviewer 2: ```suggestion
- Simple Example - A basic example of a service that is in charge of processing user signups. Great place to start learning AsyncAPI.
```

# Navigation panel

This guide will walk you through the functions of each button in navigation panel. 

# Navigation Button

The first button in the navigation panel is the Navigation button. It is one of the default active buttons.

Navigation button is divided into four parts. It provides the following information:

- Servers
- Operations
- Messages
- Schemas

## Servers

> The Servers section displays the name and information of the server you are using to build your AsyncAPI.

Reviewer 3: I suggest to just link or reuse to existing concept definitions as we already have them.

Author: I could not find the existing definition link.

Reviewer 3: For example server > https://www.asyncapi.com/docs/concepts/server

Reviewer: Adding a link as suggested.

> The Servers section displays the name and information of the server you are using to build your AsyncAPI.

Reviewer: ```suggestion
The Servers section displays the name and information of the [server]( https://www.asyncapi.com/docs/concepts/server) you are using to build your AsyncAPI.
```

## Operations

The Operations section lists all the operations that are a part of the AsyncAPI. It adds the perfix `PUB`  for publisher and `SUB` for subscriber to differentiate between both the operations.

## Messages

This Messages section displays the messages stored in `components.messages` section. This does not mean that every message in the component is used.  It can be defined but it may not used by references in the `operation` field.

## Schemas

The Schemas displays the schemas stored in the `components.schemas` section. This does not mean that every schema is used in the document. You can have it defined but it may not be used by references in the `message.payload` field.

# Editor

> The Editor section is where you edit the AsyncAPI code. As you edit, you will see the changes reflect dynamically in the HTML preview and Block Visualizer section.

Reviewer 3: `AsyncAPI code` -> `AsyncAPI file`

- hints (in green)
by its title. The description of the diagnostics with line number is displayed in a table.

To the right side of the title bar there are three indicators of the Diagnostics window: 

- Valid/Invalid - It is green if the syntax is green and red when invalid.
- Autosave - It shows whether Autosave is on or off.
- Type - Shows the language in which the API is written: `JSON` or `YAML`.

Below the title bar there is a control panel. You can the buttons to:

- Hide and show erros
- Hide and show warnings
- Hide and show information messages
- Hide and show hints

There is also search bar to search that aids in finding issues by their names. You will see Settings icon to the right. It will take you to the governance of Diagnostic's settings window.

# HTML Preview

This section displays the preview of everything you have written in the code editor. It uses different fonts, colors, and styles for readability. Each section has a different card. 

It starts with displaying the title, version, description of the API. It renders the formatting too. You can expand each element too see the `key: value` pair of each element, type, payload, and message.

# Blocks visualizer

> Blocks visualizer is the USP of this product. It gives you a quick overview of your API. Unlike HTML preview, Blocks visualizer gives a diagrammatic visual of your API. 

Reviewer 3: Spell out the full term of `USP` Acronym.
e.g `unique selling point (USP)`

We could completely remove this sentence : `Blocks visualizer is the USP of this product` as it's not useful in the documentation context.

- To bottom left of the window there are four controls: zoom in, zoom out, full-screen, toogle interactivity. You zoom the cards in and out or view them in full-screen mode. To prevent the cards from shifting use the toggle interactivity control.

# New File

To create a new file you can pick one of the available templates.

## Templates

There are 5 available to quick start Studio. 

- Simple Example - A basic example of a service that is in charge of processing user signups. Great place to start learning AsyncAPI.

- Apache Kafka - A framework implementation of a software bus using stream-processing. Open Source developed by the Apache Software Foundation.

- WebSocket - A framework implementation of a software bus using stream-processing. Open Source developed by the Apache Software Foundation.

- MQTT - A protocol for fetching resources. It is the foundation of any data exchange on the Web and it is a client-server protocol.

- HTTP - A protocol for fetching resources. It is the foundation of any data exchange on the Web and it is a client-server protocol.

## Real World Examples

> Studio provides two real world examples templates to help you get started.

Reviewer 3: This sentence can be rephrased or grammatically simplified. typo error : `real-world example templates`

## Messages

> This Messages section displays the messages stored in `components.messages` section.

Author: Hey @magicmatatjahu 

I have updated the description for `Messages` and `Schemas`.

Do you want me to add the following line to the description as well?

> - we can have it defined but not used by references in the operation fields

Reviewer 2: Yep, you should add this note, that messages and schemas are defined in components but it can (but not must) be used :)
